### ISSUE TITLE
My results look worse

### ISSUE THREAD

**Issue by MZ-Sun** [knoveleng/open-rs#9]
I was able to follow the scripts to train the model with exp3 settings, however, my performance on benchmarks are lower than the paper reported (See below, each is after 500 total global steps), even worse than baseline . What could be possible reasons?

AIME 24		          MATH500 		AMC 23		         OlympiadBench

0.33 pm 0.08		0.83 pm 0.01 		 0.65 pm 0.07 		 0.5 pm 0.02

Also, does the code shuffle the dataset before training for every epoch? Maybe that's the reason?

**Comment by QuyAnh2005** [comment]
@MZ-Sun Where did you save checkpoints? Let's try to taking checkpoint at step 50 or 100. 

If you uploaded them on huggingface, you can take a specific checkpoint easily by revision id.

**Comment by QuyAnh2005** [comment]
In case of full training, you should extend completion length instead of 3584 as in the paper

**Comment by MZ-Sun** [comment]
> In case of full training, you should extend completion length instead of 3584 as in the paper

You mean I shouldn't set completion length to 3584 in my training scripts? Why not.


Also, does the code shuffle the dataset before training for every epoch? Could data randomisation be the reason?

**Comment by QuyAnh2005** [comment]
@MZ-Sun 

Question 1: *You mean I shouldn't set completion length to 3584 in my training scripts? Why not*
Answer: As observed in Experiment 1 (Insight 1), using a completion length of 3584 is effective only for the first 50–150 steps. After 200 steps, the model's performance starts to degrade because this length restricts its reasoning ability. Therefore, when presenting results, we typically compare at step 50 or 100. As I mentioned, if you want to train the model fully, you should either increase the completion length or adopt a multi-stage training approach with varying lengths.

Question 2: *Also, does the code shuffle the dataset before training for every epoch? Could data randomisation be the reason?*
Answer: From trl, the code shuffle the dataset before training for every epoch. While randomization could be a contributing factor, I believe the limitation of completion length plays a more significant role.

**Comment by MZ-Sun** [comment]
> [@MZ-Sun](https://github.com/MZ-Sun)
> 
> Question 1: _You mean I shouldn't set completion length to 3584 in my training scripts? Why not_ Answer: As observed in Experiment 1 (Insight 1), using a completion length of 3584 is effective only for the first 50–150 steps. After 200 steps, the model's performance starts to degrade because this length restricts its reasoning ability. Therefore, when presenting results, we typically compare at step 50 or 100. As I mentioned, if you want to train the model fully, you should either increase the completion length or adopt a multi-stage training approach with varying lengths.
> 
> Question 2: _Also, does the code shuffle the dataset before training for every epoch? Could data randomisation be the reason?_ Answer: From trl, the code shuffle the dataset before training for every epoch. While randomization could be a contributing factor, I believe the limitation of completion length plays a more significant role.

Thank you for the reply. My question is solved for now.
